test/parallel/test-http-client-reject-unexpected-agent.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const baseOptions = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+};
+
+const failingAgentOptions = [
+  true,
+  'agent',
+  {},
+];
+
+const acceptableAgentOptions = [
+  false,
+  undefined,
+  null,
+  new http.Agent(),
+];
+
+
+const server = http.createServer((req, res) => {
+  res.end('hello');
+});
+
+const requestsExpected = failingAgentOptions.length +
+                         acceptableAgentOptions.length;
+let numberOfRequests = 0;
+
+function createRequest(agent) {
+  numberOfRequests++;
+  const options = Object.assign(baseOptions, {agent});
+  const request = http.request(options);
+  let closeServer = false;
+  request.end();
+  if (numberOfRequests === requestsExpected) {
+    closeServer = true;
+  }
+  request.on('response', common.mustCall(() => {
+    if (closeServer) server.close();
```

Comment:
Couldn't you just increment `numberOfRequests` here, and remove the `closeServer` variable. Also, can you move `server.close()` to a new line.